posthog/src/main/java/com/posthog/PostHog.kt

```diff
@@ -484,27 +506,30 @@ public class PostHog private constructor(
             config?.logger?.log("identify called with invalid anonymousId: $anonymousId.")
         }
 
-        capture(
-            "\$identify",
-            distinctId = distinctId,
-            properties = props,
-            userProperties = userProperties,
-            userPropertiesSetOnce = userPropertiesSetOnce,
-        )
+        if (previousDistinctId != distinctId && !this.isIdentified) {
```

Comment:
I think the behavior here is different, see the JS SDK https://github.com/PostHog/posthog-js/blob/e8a645134a03477d29ab6839ae7f10fcd6e4157e/src/posthog-core.ts#L1316-L1354
if it's the very same `distinctId` again won't capture anything (and capture a $set event if needed)
if it's a new distinctId, clear all the cache from the previous distinctId, and identify again with the new distinctId, reload feature flags, etc

Reply:
Isn't it nearly the same behavior, except for the `setPersonProperties` and `setPersonPropertiesForFlags`, which are not yet implemented here.

Reply:
nvm, I think I misread the PR, you are right.